From 9d96be3f6c0fc8f8281f78c956f8e90d18a4bc4e Mon Sep 17 00:00:00 2001
From: hariprasad <hkelam@marvell.com>
Date: Thu, 4 Jul 2019 16:50:16 +0530
Subject: [PATCH 269/767] octeontx2-af: Release mutex resource

commit 08330f80f08b96720881a2e9197d8d8cd013bff4 from
git@git.assembla.com:cavium/WindRiver.linux.git

When all mcam entries are free current code returns
without releasing mutex.This patch fixes it

Change-Id: I44fc8504c0a84c929947fdd09d10b999b569e7a8
Signed-off-by: hariprasad <hkelam@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/12129
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu_debugfs.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu_debugfs.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_debugfs.c
index 187553551c20..e48530392d90 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu_debugfs.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu_debugfs.c
@@ -1416,8 +1416,10 @@ static int rvu_dbg_npc_mcam_info_display(struct seq_file *filp, void *unsued)
 	seq_printf(filp, "\t\t Available \t: %d\n",
 		   rvu_rsrc_free_count(&mcam->counters));
 
-	if (mcam->bmap_entries == mcam->bmap_fcnt)
+	if (mcam->bmap_entries == mcam->bmap_fcnt) {
+		mutex_unlock(&mcam->lock);
 		return 0;
+	}
 
 	seq_puts(filp, "\n\t\t Current allocation\n");
 	seq_puts(filp, "\t\t====================\n");
-- 
2.31.1

